### ISSUE TITLE
Hiding values in view results

### ISSUE THREAD

**Issue by jamesrkg** [Quantipy/quantipy#18]
We need a way to hide specific values in the result of views.

For example, given the following view:

```python
Question            gender  
Values              @
Question    Values  
q1          1       7.201447
            2       9.64925
            3       55.662665
            4       72.663656
            5       4.704566
            6       11.496034
            7       21.69883
            8       3.178551
            9       0.099151
            96      2.199429
            98      2.497416
            99      8.949005
```

We need a way to hide selected values in the index. For example, in this case the values 96, 98 and 99 which represent some form of non-response like "Don't know", "Prefer not to say", "None of these" have been hidden.

```python
Question            gender          
Values              @
Question    Values          
q1          4       72.663656
            3       55.662665
            7       21.69883
            6       11.496034
            2       9.64925
            1       7.201447
            5       4.704566
            8       3.178551
            9       0.099151
```

I'll implement this using column meta tags that are picked up by paint_dataframe.

**Comment by jamesrkg** [comment]
I've implemented this in a new meta object called ``rules``, using an option called ``dropx``.

The ``rules`` object is a dict with ``x`` and ``y`` keys for axis-specific rules application. Internally ``dropx`` results in a call to a function of the same name. The contents of the ``slicex`` object becomes that function call's kwargs.

```python
def dropx(df, values):
    """
    Return df after dropping values from the index.
    
    Assuming a Quantipy-style view result this function drops index
    rows indicated by values and returns the result.
    
    Parameters
    ----------
    df : pandas.DataFrame
        The dataframe that should have some index rows dropped.
    values : list-like
        A list of index values that should be dropped from the index.
    
    Returns
    -------
    df : list
        The edited dataframe. 
    """
```

Given the following crosstab:

```python
print crosstab(meta, data, 'income', 'income_cb')

Question                income_cb                                                                        
Values                        All    1    2    3    4    5    6   7   8   9   10   11   12  901  902  903
Question    Values                                                                                       
income      All              2078  425  204  187  146  119  103  89  68  59  226  101  351  425  848  353
            1                 425  425    0    0    0    0    0   0   0   0    0    0    0  425    0    0
            2                 204    0  204    0    0    0    0   0   0   0    0    0    0    0  204    0
            3                 187    0    0  187    0    0    0   0   0   0    0    0    0    0  187    0
            4                 146    0    0    0  146    0    0   0   0   0    0    0    0    0  146    0
            5                 119    0    0    0    0  119    0   0   0   0    0    0    0    0  119    0
            6                 103    0    0    0    0    0  103   0   0   0    0    0    0    0  103    0
            7                  89    0    0    0    0    0    0  89   0   0    0    0    0    0   89    0
            8                  68    0    0    0    0    0    0   0  68   0    0    0    0    0    0   68
            9                  59    0    0    0    0    0    0   0   0  59    0    0    0    0    0   59
            10                226    0    0    0    0    0    0   0   0   0  226    0    0    0    0  226
            11                101    0    0    0    0    0    0   0   0   0    0  101    0    0    0    0
            12                351    0    0    0    0    0    0   0   0   0    0    0  351    0    0    0
```

Set up the ``dropx`` ``rules``:

```python
meta['columns']['AMA_qincome']['rules'] = {
    'x': {'dropx': {'values': [1, 3, 5, 7, 9, 11]}}
}

meta['columns']['AMA_qincome_cb']['rules'] = {
    'y': {'dropx': {'values': [12, 10, 4, 6, 2, 901, 902, 903]}}
}
```

Apply the ``rules`` using crosstab:

```python
print crosstab(meta, data, 'income', 'income_cb', rules=True)


Question                income_cb                                
Values                        All    1    3    5   7   8   9   11
Question    Values                                               
income      All              2078  425  187  119  89  68  59  101
            2                 204    0    0    0   0   0   0    0
            4                 146    0    0    0   0   0   0    0
            6                 103    0    0    0   0   0   0    0
            8                  68    0    0    0   0  68   0    0
            10                226    0    0    0   0   0   0    0
            12                351    0    0    0   0   0   0    0
```

**Comment by jamesrkg** [comment]
The ``rules`` object also supports ``slicex`` (#19) and ``sortx`` (#15), all of which may be used in any combination.

**Comment by jamesrkg** [comment]
When drawing chains out of a stack, rules can be applied to what is extracted in the same way:

```python
chains = stack.get_chain(..., rules=True)
```

``rules`` will currently only be applied to chain items if the parameter ``post_process`` is also ``True`` (which it is by default).

**Comment by jamesrkg** [comment]
Resolved by #40
